src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs

```diff
@@ -188,7 +192,7 @@ private unsafe void FlushInternal(bool endOfRequest, ArraySegment<byte> data = n
         }
     }
 
-    private List<GCHandle> PinDataBuffers(bool endOfRequest, ArraySegment<byte> data, out HttpApiTypes.HTTP_DATA_CHUNK[] dataChunks)
+    private List<GCHandle> BuildDataChunks(ref UnmanagedBufferAllocator allocator, bool endOfRequest, ArraySegment<byte> data, out HttpApiTypes.HTTP_DATA_CHUNK[] dataChunks)
     {
         var pins = new List<GCHandle>();
```

Comment:
We should not need growable `List<GCHandle>' given the other changes. We know exactly how many GCHandles we are going to needed. We can allocate `GCHandle[]` of the exact size, similar how `HTTP_DATA_CHUNK[]` is allocated.